langchain4j-pinecone/src/main/java/dev/langchain4j/store/embedding/pinecone/PineconeEmbeddingStore.java

```diff
@@ -136,7 +138,10 @@ private void addAllInternal(List<String> ids, List<Embedding> embeddings, List<T
                 vectorBuilder.setMetadata(Struct.newBuilder()
                         .putFields(metadataTextKey, Value.newBuilder()
                                 .setStringValue(textSegments.get(i).text())
-                                .build()));
+                                .build())
+                        .putAllFields(textSegments.get(i).metadata().asMap().entrySet()
+                                .stream()
+                                .collect(Collectors.toMap(Map.Entry::getKey, e -> Value.newBuilder().setStringValue(e.getValue()).build()))));
```

Comment:
With this change, metadata is stored, but not retrieved during read. Please update `PineconeEmbeddingStoreIT` to extend `EmbeddingStoreIT` (instead of `EmbeddingStoreWithoutMetadataIT`). `should_add_embedding_with_segment_with_metadata` will fail.

Reply:
Thanks, I'll try to get to it today

Reply:
Changed as requested - is that it? Sorry, I am not familiar with Java testing

Reply:
All my tests succeed when I run `./mvnw test` except for Neo4J which I doubt is related

Reply:
Hi @nsadeh, `should_add_embedding_with_segment_with_metadata` test still fails because there is no code that reads stored metadata key-value pairs when retrieving embeddings. You have added code for writing metadata, but not for reading back.

Reply:
@langchain4j latest commit should do it

Reply:
@nsadeh 3 tests are now failing:
`should_add_embedding_with_segment`
`should_add_multiple_embeddings_with_segments`
`should_add_embedding_with_segment_with_metadata`

Reply:
I am going to need a lot more hand-holding on the tests here.

When I run `./mvnw test`, the pinecone tests succeed.

When I run these tests specifically from IntelliJ, they fail because they're missing an API key. I haven't been able to find the test api key in the repo.

What's the SOP here for running pinecone tests?